components/lcd/esp_lcd_st7796/CMakeLists.txt

```diff
@@ -1,5 +1,6 @@
-idf_component_register(SRCS "esp_lcd_st7796.c"
+idf_component_register(SRCS "esp_lcd_st7796.c" "esp_lcd_st7796_general.c" "esp_lcd_st7796_mipi.c"
```

Comment:
should only add the `"esp_lcd_st7796_mipi.c"` when `CONFIG_SOC_MIPI_DSI_SUPPORTED` is true.

Reply:
Similarly, this is also written this way to facilitate the migration of the component to Arduino.